pkg/plugins/storage/plugin.go

```diff
@@ -25,6 +25,9 @@ const (
 
 func (op Operation) String() string {
 	// The order of this array must match the iota order above.
+	if op != WRITE && op != READ {
```

Comment:
If an unknown operation is passed here it should probably return "UNKNOWN" or something similar instead of defaulting to read. Otherwise, error messages could be misleading when they claim a read operation failed when really it was some other unsupported operation. (Whatever you do here we should do the same for the existing AWS S3 implementation).

Reply:
maybe use: https://pkg.go.dev/golang.org/x/tools/cmd/stringer